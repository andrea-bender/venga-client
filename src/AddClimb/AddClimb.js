import React, {Component} from 'react';
import ClimbsContext from '../ClimbsContext';
import config from '../config';
import './AddClimb.css';

const Required = () => (
  <span className='AddClimb_required'> * </span>
);
export default class AddClimb extends Component {
  static defaultProps = {
    history: {
      push: () => { }
    }
  };

  static contextType = ClimbsContext;

  handleSubmit = e => {
    e.preventDefault();

    const { name, location, description, type, grade, rating } = e.target; 
    const newClimb = {
      name: name.value,
      location: location.value,
      description: description.value,
      type: type.value,
      grade: grade.value,
      rating: rating.value
    };

    fetch(`${config.API_ENDPOINT}/climbs`, {
      method: 'POST',
      body: JSON.stringify(newClimb),
      headers: {
        'content-type': 'application/json',
      }
    })
    .then(res => {
      if(!res.ok) {
        return res.json().then(error => Promise.reject(error))
      }
      return res.json()
    })
    .then(data => {
      name.value = ''
      location.value = ''
      description.value = ''
      type.value = ''
      grade.value = ''
      rating.value = 0
      this.context.AddClimb(data)
      this.props.history.push('/')
    })
    .catch(error => {
      this.setState({error})
    })

  };

  render() {
    return(
      <section className='AddClimb'>
        <h3> Add a new climb! </h3>
        
        <form
          className='AddClimb_form'
          onSubmit={this.handleSubmit}
        >
          <div>
            <label htmlFor="Name">Name: {' '} <Required /></label> <br />
            <input type="text" id="name" name="Name" placeholder="Name of climb" required />
          </div>

          <div>
            <label htmlFor="Location">Location: {' '} <Required /></label> <br />
            <input type="text" id="location" name="location" required />
          </div>

          <div>
            <label htmlFor="Description"> Description: {' '} <Required /></label> <br />
            <textarea id="description" name="description" rows="6" cols="45" required></textarea>
          </div>

          <div>
            <label htmlFor="Grade">Grade: {' '} <Required /></label> <br />
            <input type="text" id="grade" name="grade" required />
          </div>

          <div>
            <label for="type">Type</label>
              <select id="type">
                <option value="Boulder">Boulder</option>
                <option value="Sport">Sport</option>
                <option value="Traditional">Traditional</option>
                <option value="Ice">Ice</option>
                <option value="Other">Other</option>
              </select>
          </div>

          <div className='rating'>
            <label>
              <input type="radio" name="stars" value="1" />
              <span class="icon">★</span>
            </label>
            <label>
              <input type="radio" name="stars" value="2" />
              <span class="icon">★</span>
              <span class="icon">★</span>
            </label>
            <label>
              <input type="radio" name="stars" value="3" />
              <span class="icon">★</span>
              <span class="icon">★</span>
              <span class="icon">★</span>   
            </label>
            <label>
              <input type="radio" name="stars" value="4" />
              <span class="icon">★</span>
              <span class="icon">★</span>
              <span class="icon">★</span>
              <span class="icon">★</span>
            </label>
            <label>
              <input type="radio" name="stars" value="5" />
              <span class="icon">★</span>
              <span class="icon">★</span>
              <span class="icon">★</span>
              <span class="icon">★</span>
              <span class="icon">★</span>
            </label>
          </div>

          <div>
            <button type="submit"> Add your climb </button>
          </div>
        </form>
      </section>
    )
  }
  
};