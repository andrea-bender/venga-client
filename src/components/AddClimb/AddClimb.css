form{
  margin-left:auto;
  margin-right: auto;
}

 @media only screen and (min-width: 300px) {
  .AddClimb{
    background-color: black;
    color: white;
    height: 100vh;
    font-weight: 600;
  }
  .AddClimb h3{
    text-align: center;
  }
  .add-climb-top .logo-img{
    width: 10em;
  }
  .AddClimb_form{
  background-image: url(../../assets/bg3.jpeg); 
  background-position: center; 
  background-size: cover;
  height: 70vh;
  border-radius: 2px;
  text-align: center;
  }
  .AddClimb_form div{
    padding-bottom: 1em;
  }
  .AddClimb_form select{
    background-color: white;
    height:2em
  }
  .AddClimb_form label{
    padding-right: .5em
  }
  .AddClimb_form button{
    background-color: white;
    padding:.5em;
    margin-left: 1em;
    width: 7em;
  }
}
@media only screen and (min-width: 750px) {
  .add-climb-top .logo-img {
      width: 21em;
  }
  .AddClimb h3 {
    font-size: 3em;
  }
  .AddClimb_form label{
    font-size: 2.2em;
  }
  .AddClimb_form input{
    font-size: 1.5em;
  }
  .AddClimb_form textarea{
    font-size: 1.5em;
  }
  
}
@media only screen and (min-width: 1024px) {
  .AddClimb_form label {
    font-size: 1.2em;
  }
  textarea {
    height: 6em;
    width: 14em;
  }
  .AddClimb_form button:hover{
    cursor: pointer;
    color: rgb(0, 173, 173);
  }
}