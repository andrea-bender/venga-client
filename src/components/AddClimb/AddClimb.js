import React, {Component} from 'react';
import { Link } from 'react-router-dom';
import ClimbsContext from '../../contexts/ClimbsContext';
import config from '../../config';
import './AddClimb.css';

const Required = () => (
  <span className='AddClimb_required'> * </span>
);
export default class AddClimb extends Component {
  static defaultProps = {
    history: {
      push: () => { }
    }
  };

  static contextType = ClimbsContext;

  handleSubmit = e => {
    e.preventDefault();

    const { name, location, description, type, grade, rating } = e.target; 
    const newClimb = {
      name: name.value,
      location: location.value,
      description: description.value,
      type: type.value,
      grade: grade.value,
      rating: rating.value
    };
    fetch(`${config.API_ENDPOINT}/climbs`, {
      method: 'POST',
      body: JSON.stringify(newClimb),
      headers: {
        'content-type': 'application/json',
      }
    })
    .then(res => {
      if(!res.ok) {
        return res.json().then(error => Promise.reject(error))
      }
      return res.json()
    })
    .then(data => {
      name.value = ''
      location.value = ''
      description.value = ''
      type.value = ''
      grade.value = ''
      rating.value = ''
      this.context.AddClimb(data)
      this.props.history.push('/climbs')
    })
    .catch(error => {
      this.setState({error})
    })

  };

  render() {
    return(
      <section className='AddClimb'>
        <div className='add-climb-top'>
        <Link className="logo-img" to="/">
          <img
                className="logo-img"
                src={require('../../assets/logo1.png')}
                alt={'venga logo'}
                width={'200px'}
            />
            </Link>
        </div>
        <h3> Add a new climb! </h3>
        
        <form
          className='AddClimb_form'
          onSubmit={this.handleSubmit}
        >
          <div>
            <label htmlFor="Name">Name: {' '} <Required /></label> <br />
            <input type="text" id="name" name="Name" placeholder="Name of climb" required />
          </div>

          <div>
            <label htmlFor="Location">Location: {' '} <Required /></label> <br />
            <input type="text" id="location" name="location" placeholder="Where is it located"required />
          </div>

          

          <div>
            <label htmlFor="Grade">Grade: {' '} <Required /></label> <br />
            <input type="text" id="grade" name="grade" placeholder="Difficulty"required />
          </div>

          <div>
            <label htmlFor="type">Type:</label>
              <select id="type">
                <option value='0'>Select Type</option>
                <option value="Boulder">Boulder</option>
                <option value="Sport">Sport</option>
                <option value="Traditional">Traditional</option>
                <option value="Ice">Ice</option>
                <option value="Other">Other</option>
              </select>
          </div>
          <div>
            <label htmlFor="rating">Rating:</label>
              <select id="rating">
                <option value="0">Select Rating</option>
                <option value="1">★☆☆☆☆</option>
                <option value="2">★★☆☆☆</option>
                <option value="3">★★★☆☆</option>
                <option value="4">★★★★☆</option>
                <option value="5">★★★★★</option>
              </select>
          </div>
          <div>
            <label htmlFor="Description"> Description: {' '} <Required /></label> <br />
            <textarea id="description" name="description" rows="6" cols="40" placeholder="Write your opinion..."required></textarea>
          </div>
          <div>
            <button type="submit"> Add Climb </button>
            <Link to={'/'}><button type='button'> Back </button></Link>

          </div>
        </form>
      </section>
    )
  }
  
};