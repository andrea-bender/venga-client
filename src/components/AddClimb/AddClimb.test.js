import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddClimb from './AddClimb'

describe(`AddClimb component`, () => {
  const stubAddClimb = [
    {
      "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "climb1",
      "location": "Test location",
      "description": "test desc",
      "grade": 'test grade',
      'type':'Boulder',
      'rating':'5'
    },
    {
      "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "climb2",
      "location": "Test location 2",
      "description": "test desc 2",
      "grade": 'test grade 2',
      'type':'Sport',
      'rating':'3'
    }
  ]

  it('renders the complete form', () => {
    const context = { rescipes: stubAddClimb }
    const wrapper = shallow(<AddClimb />, context)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
})