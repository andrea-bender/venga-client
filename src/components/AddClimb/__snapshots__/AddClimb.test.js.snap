// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddClimb component renders the complete form 1`] = `
<section
  className="AddClimb"
>
  <div
    className="add-climb-top"
  >
    <Link
      className="logo-img"
      to="/"
    >
      <img
        alt="venga logo"
        className="logo-img"
        src="logo1.png"
        width="200px"
      />
    </Link>
  </div>
  <h3>
     Add a new climb! 
  </h3>
  <form
    className="AddClimb_form"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="Name"
      >
        Name: 
         
         
        <Required />
      </label>
       
      <br />
      <input
        id="name"
        name="Name"
        placeholder="Name of climb"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="Location"
      >
        Location: 
         
         
        <Required />
      </label>
       
      <br />
      <input
        id="location"
        name="location"
        placeholder="Where is it located"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="Grade"
      >
        Grade: 
         
         
        <Required />
      </label>
       
      <br />
      <input
        id="grade"
        name="grade"
        placeholder="Difficulty"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="type"
      >
        Type:
      </label>
      <select
        id="type"
      >
        <option
          value="0"
        >
          Select Type
        </option>
        <option
          value="Boulder"
        >
          Boulder
        </option>
        <option
          value="Sport"
        >
          Sport
        </option>
        <option
          value="Traditional"
        >
          Traditional
        </option>
        <option
          value="Ice"
        >
          Ice
        </option>
        <option
          value="Other"
        >
          Other
        </option>
      </select>
    </div>
    <div>
      <label
        htmlFor="rating"
      >
        Rating:
      </label>
      <select
        id="rating"
      >
        <option
          value="0"
        >
          Select Rating
        </option>
        <option
          value="1"
        >
          ★☆☆☆☆
        </option>
        <option
          value="2"
        >
          ★★☆☆☆
        </option>
        <option
          value="3"
        >
          ★★★☆☆
        </option>
        <option
          value="4"
        >
          ★★★★☆
        </option>
        <option
          value="5"
        >
          ★★★★★
        </option>
      </select>
    </div>
    <div>
      <label
        htmlFor="Description"
      >
         Description: 
         
         
        <Required />
      </label>
       
      <br />
      <textarea
        cols="40"
        id="description"
        name="description"
        placeholder="Write your opinion..."
        required={true}
        rows="6"
      />
    </div>
    <div>
      <button
        type="submit"
      >
         Add Climb 
      </button>
      <Link
        to="/"
      >
        <button
          type="button"
        >
           Back 
        </button>
      </Link>
    </div>
  </form>
</section>
`;
